fee/src/lib.rs

```diff
@@ -28,13 +28,140 @@ pub fn calculate_fee_details(
     if zero_fees_for_test {
         return FeeDetails::default();
     }
-    let signature_fee = message
-        .num_total_signatures()
-        .saturating_mul(lamports_per_signature);
+
+    calculate_fee_details_internal(
+        SignatureCounts::from(message),
+        lamports_per_signature,
+        prioritization_fee,
+        remove_rounding_in_fee_calculation,
+    )
+}
+
+/// This function has the actual fee calculation.
+fn calculate_fee_details_internal(
+    SignatureCounts {
+        num_transaction_signatures,
+        num_ed25519_signatures,
+        num_secp256k1_signatures,
+    }: SignatureCounts,
+    lamports_per_signature: u64,
+    prioritization_fee: u64,
+    remove_rounding_in_fee_calculation: bool,
+) -> FeeDetails {
+    let signature_fee = (num_transaction_signatures
+        .saturating_add(num_ed25519_signatures)
+        .saturating_add(num_secp256k1_signatures))
+    .saturating_mul(lamports_per_signature);
 
     FeeDetails::new(
         signature_fee,
         prioritization_fee,
         remove_rounding_in_fee_calculation,
     )
 }
+
+struct SignatureCounts {
```

Comment:
see follow-up PR in #4472 which adds the secpr1 signature count behind a feature-gate.